Update Xamarin tutorial (#40)
DevExpress/XPO: Tutorials/Xamarin.Forms/README.md

## Overview

This tutorial demonstrates how to create a Xamarin mobile application for iOS and Android. The application uses the [DevExpress ORM Library (XPO)](https://www.devexpress.com/products/net/orm/) to access and manage data and stores data in an SQLite database. 

<p align="center">
  <img width="540" src="/Tutorials/images/Xamarin.Forms/8.1.png">
</p>

<p align="center">
  <img width="540" src="/Tutorials/images/Xamarin.Forms/8.2.png">
</p>

## Prerequisites

* [Visual Studio 2019](https://visualstudio.microsoft.com/vs/) with the **Mobile development with .NET** workload.
* (Optional) A paired Mac to build the application on iOS.

## Step 1: Create a Mobile App

1. Open Visual Studio and create a new project.
2. Search for the **Mobile App (Xamarin Forms)** template. 

    ![](/Tutorials/images/Xamarin.Forms/1.1.png)

    Click **Next** to proceed.

3. Set the project name to **XamarinFormsDemo** and click **Create**.

    ![](/Tutorials/images/Xamarin.Forms/1.2.png)

4. Select the **Master-Detail** application template and click **OK**.

    ![](/Tutorials/images/Xamarin.Forms/1.3.png)

> 5. In the **Solution Explorer**, remove unnecessary files:

Reviewer: ```suggestion
5. In the **Solution Explorer**, remove the unnecessary files:
```

    * *Models\HomeMenuItem.cs*
    * *Services\MockDataStore.cs*
    * *Views\MenuPage.xaml*

    ![](/Tutorials/images/Xamarin.Forms/1.4.png) 

For more information, see the following:

- [Build your first Xamarin.Forms App](https://docs.microsoft.com/en-us/xamarin/get-started/first-app/)
- [Xamarin.Forms - Quick Starts](https://docs.microsoft.com/en-us/xamarin/get-started/quickstarts/)

## Step 2: Add the NuGet Packages

> The application you build in this tutorial requires installing the following NuGet Packages:

Reviewer: ```suggestion
The application you build in this tutorial requires the following NuGet Packages:
```

- DevExpress.Xpo
- Microsoft.Data.Sqlite

> From the Visual Studio **Tools** menu, select **NuGet Package Manager > Package Manager Console**.

Reviewer: ```suggestion
From Visual Studio's **Tools** menu, select **NuGet Package Manager > Package Manager Console**.
```

Make sure **Package source** is set to **All** or **nuget.org** and run the following commands: 

```console
Install-Package DevExpress.Xpo
```

```console
Install-Package Microsoft.Data.Sqlite
```

## Step 3: Create ORM Data Model and Seed Initial Data

> 1. The project already contains the _Item_ class (the _"/Models/Item.cs"_ file). Change its code as follows:

Reviewer: ```suggestion
1. Change the project's _Item_ class (the _"/Models/Item.cs"_ file) as follows:
```

    ```cs
    using DevExpress.Xpo;

    namespace XamarinFormsDemo.Models {
        [Persistent]
        public class Item {
            [Key]
            public string Id { get; set; }
            [Size(256)]
            public string Text { get; set; }
            [Size(SizeAttribute.Unlimited)]
            public string Description { get; set; }
        }
    }
    ```

2. In the **Solution Explorer**, create the **Core** folder and add the *XpoHelper.cs* code file. 

>     Replace its content with the code copied from the [corresponding file](https://github.com/DevExpress/XPO/tree/master/Tutorials/Xamarin.Forms/XamarinFormsDemo/Core/XpoHelper.cs). This file contains the `XpoHelper` class.

Reviewer: "the [corresponding file]" Which corresponding file? Do you mean

Reviewer: ```suggestion
    Replace its content with the code copied from the [file](https://github.com/DevExpress/XPO/tree/master/Tutorials/Xamarin.Forms/XamarinFormsDemo/Core/XpoHelper.cs) that contains the `XpoHelper` class.
```

    
    The `XpoHelper` class creates and saves initial data in the database. 

    ![](/Tutorials/images/Xamarin.Forms/3.2.png) 

For more information, see the following:

* [Add Persistence to an Existing Hierarchy](https://docs.devexpress.com/XPO/2108/fundamentals/adding-persistence-to-an-existing-hierarchy)
* [Object Relational Mapping](https://docs.devexpress.com/XPO/2017/feature-center/object-relational-mapping)
* [Persisting Objects](https://docs.devexpress.com/XPO/2025/feature-center/data-exchange-and-manipulation/persisting-objects)

## Step 4: Implement CRUD Operations and Connect App to SQLite

> 1. In the **Solution Explorer**, select the **Services** folder and add the *XpoDataStore.cs* code file. Replace the code with the code from the [corresponding file](https://github.com/DevExpress/XPO/tree/master/Tutorials/Xamarin.Forms/XamarinFormsDemo/Services/XpoDataStore.cs).

Reviewer: Same here
